lib/index.js

```diff
@@ -87,6 +90,18 @@ exports.request = function (method, url, options, callback, _trace) {
             req.abort();
         }
 
+        if (logger.active) {
```

Comment:
lets simplify this to `logger.log(err, url, options, req, res);` and make the logger wrap the res stream when it needs to output details from it.  What I mean is, inside of `log` if we are going to need to read the res then do it then... but move that logic into logger.